queue.go

```diff
@@ -0,0 +1,165 @@
+	if len(errs) != 0 {
+		for uuid, err := range errs {
+			logrus.Errorf("Couldn't build message for image-set transactionId=%v uuid=%v %v", tid, uuid, err)
+		}
+	}
+	for uuid, msg := range msgs {
+		err = q.messageProducer.SendMessage("", msg)
+		if err != nil {
+			logrus.Errorf("Error sending transformed message to queue transactionId=%v uuid=%v %v", tid, uuid, err)
+			continue
+		}
+		logrus.Infof("Mapped and sent for uuid=%v transactionId=%v", uuid, tid)
+	}
+}
+
+func (q defaultQueue) buildMessages(imageSets []JSONImageSet, lastModified string, tid string) (map[string]producer.Message, map[string]error) {
+	errs := make(map[string]error, 0)
+	msgs := make(map[string]producer.Message, 0)
+	for _, imageSet := range imageSets {
+		msg, err := q.buildMessage(imageSet, lastModified, tid)
+		if err != nil {
+			errs[imageSet.UUID] = err
+			continue
+		}
+		msgs[imageSet.UUID] = msg
+	}
+	return msgs, errs
+}
+
+func (q defaultQueue) buildMessage(imageSet JSONImageSet, lastModified, pubRef string) (producer.Message, error) {
+	headers := map[string]string{
+		"X-Request-Id":      pubRef,
+		"Message-Timestamp": lastModified,
+		"Message-Id":        gouuid.NewV4().String(),
+		"Message-Type":      "cms-content-published",
+		"Content-Type":      "application/json",
+		"Origin-System-Id":  methodeSystemOrigin,
+	}
+	body := publicationMessageBody{
+		ContentURI:   contentURIBase + imageSet.UUID,
+		Payload:      imageSet,
+		LastModified: lastModified,
+	}
+	marshaledBody, err := q.unsafeJSONMarshal(body)
+	if err != nil {
+		return producer.Message{}, fmt.Errorf("Couldn't marshall message body to JSON skipping message. transactionId=%v %v", pubRef, body)
+	}
+	return producer.Message{Headers: headers, Body: string(marshaledBody)}, nil
+}
+
+func (q defaultQueue) unsafeJSONMarshal(v interface{}) ([]byte, error) {
+	b, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	b = bytes.Replace(b, []byte("\\u003c"), []byte("<"), -1)
+	b = bytes.Replace(b, []byte("\\u003e"), []byte(">"), -1)
+	return b, nil
+}
+
+func prettyPrintConfig(consumerConfig consumer.QueueConfig, producerConfig producer.MessageProducerConfig) string {
+	return fmt.Sprintf("Config: [\n\t%s\n\t%s\n]", prettyPrintConsumerConfig(consumerConfig), prettyPrintProducerConfig(producerConfig))
+}
+
+func prettyPrintConsumerConfig(consumerConfig consumer.QueueConfig) string {
+	return fmt.Sprintf("consumerConfig: [\n\t\taddr: [%v]\n\t\tgroup: [%v]\n\t\ttopic: [%v]\n\t\treadQueueHeader: [%v]\n\t]",
+		consumerConfig.Addrs, consumerConfig.Group, consumerConfig.Topic, consumerConfig.Queue)
+}
+
+func prettyPrintProducerConfig(producerConfig producer.MessageProducerConfig) string {
+	return fmt.Sprintf("producerConfig: [\n\t\taddr: [%v]\n\t\ttopic: [%v]\n\t\twriteQueueHeader: [%v]\n\t]",
+		producerConfig.Addr, producerConfig.Topic, producerConfig.Queue)
+}
+
+func (q defaultQueue) startConsuming() {
+	var consumerWaitGroup sync.WaitGroup
```

Comment:
can't you directly use the `consumerWaitGroup` from within the `defaultQueue` ? (`q. consumerWaitGroup`)